doc/configuration.md

```diff
@@ -787,6 +787,32 @@ With the configuration excerpt:
 
 Then the `Phar::getMetadata()` will return `['application_version' => '1.0.0-dev']` array.
 
+If you specify a php file, the return contents (as array) will be included as metadata.
+
+For example, if you take the following code:
+
+```php
+<?php
+# filename:  /project-root/box-metadata.php
+class MyBox
+{
+    public static function generateMetadata()
+    {
+        return ['application_version' => '1.0.0-dev'];
+    }
+}
+return MyBox::generateMetadata();
```

Comment:
How about a comment to alert implementors this is the essential component from this file, more than anything else?

Reply:
It's not really an essential component, and should be considered as it. If file included did not return any valid data structure or callable to be use to get metadata, then the box don't be stopped to compile.

Reply:
Similar to https://github.com/box-project/box/pull/582#discussion_r777603020, I feel that if the configuration is unused, errors or warnings should be returned to the user and must not silently fail in order to have a good developer experience.